examples/hawkbit-example-mgmt-simulator/src/main/java/org/eclipse/hawkbit/mgmt/client/scenarios/ConfigurableScenario.java

```diff
@@ -166,15 +209,30 @@ private void createDistributionSets(final Scenario scenario) {
         LOGGER.info("Creating {} distribution sets", scenario.getDistributionSets());
         final byte[] artifact = generateArtifact(scenario);
 
-        distributionSetResource.createDistributionSets(new DistributionSetBuilder().name(scenario.getDsName())
-                .type("os_app").version("1.0.").buildAsList(scenario.getDistributionSets())).getBody()
-                .forEach(dsSet -> {
-                    final List<MgmtSoftwareModule> modules = addModules(scenario, dsSet, artifact);
+        final List<MgmtDistributionSet> sets = distributionSetResource
+                .createDistributionSets(new DistributionSetBuilder().name(scenario.getDsName()).type("os_app")
+                        .version("1.0.").buildAsList(scenario.getDistributionSets()))
+                .getBody();
 
-                    final SoftwareModuleAssigmentBuilder assign = new SoftwareModuleAssigmentBuilder();
-                    modules.forEach(module -> assign.id(module.getModuleId()));
-                    distributionSetResource.assignSoftwareModules(dsSet.getDsId(), assign.build());
-                });
+        sets.forEach(dsSet -> {
```

Comment:
Better extract this loop (or its content) to a private method. A proper name would be `assignSoftwareModulTo(dsSet)`